ecosystem/typescript/sdk_v2/src/internal/account.ts

```diff
@@ -148,6 +149,36 @@ export async function getResource(args: {
   return data;
 }
 
+export async function lookupOriginalAccountAddress(args: {
+  aptosConfig: AptosConfig;
+  addressOrAuthKey: HexInput;
+  options?: LedgerVersion;
+}): Promise<Hex> {
+  const { aptosConfig, addressOrAuthKey, options } = args;
+  const resource = await getResource({
+    aptosConfig,
+    accountAddress: "0x1",
+    resourceType: "0x1::account::OriginatingAddress",
+    options,
+  });
+
+  const {
+    address_map: { handle },
+  } = resource.data as any;
+  const originalAddr = await getTableItem({
+    aptosConfig,
+    handle,
+    data: {
+      key: Hex.fromHexInput({ hexInput: addressOrAuthKey }).toString(),
+      key_type: "address",
+      value_type: "address",
+    },
+    options: options,
+  });
```

Comment:
I wonder if we should migrate to using view functions, so we don't have to deal with standardization of the table address.  That'll help with address migration.

Reply:
Ahh that is a good call. I think we can migrate this after the sdkv2 alpha release.